sympy/ntheory/generate.py

```diff
@@ -246,9 +263,32 @@ def primepi(n):
     n = int(n)
     if n < 2:
         return 0
-    else:
-        return sieve.search(n)[0]
-
+    lim = int(n ** 0.5)
+    lim -= 1
+    lim = max(lim,0)
+    while lim * lim <= n:
+        lim+=1
+    lim-=1
+    arr1 = [0]*(lim+1)
+    arr2 = [0]*(lim+1)
+    for i in range(1, lim+1):
+        arr1[i] = i-1
```

Comment:
Initially,`arr1[i]` stores `phi(i,1)` ,i.e. `i-1`
Then , it stores `phi(i,2)`
Then , `phi(i,3)...`
In the end of the function it will store `phi(i,lim)`
Similarly for `arr2[i]`, stores `phi(n//i , j)` , initially, `j=1` , and in the end, `j=lim`.